src/lib/adapters/waku/index.ts

```diff
@@ -118,6 +119,33 @@ async function storeObjectStore(waku: LightNode, address: string, objectStore: O
 	await storeDocument(waku, 'objects', address, Array.from(objectStore.objects))
 }
 
+async function readBalances(waku: LightNode, address: string): Promise<BalanceState> {
+	const balancesRaw = (await readLatestDocument(waku, 'balances', address)) as TokenDb[] | undefined
+	const balances: Token[] = []
+	if (balancesRaw) {
+		balancesRaw.forEach((balanceRaw) => {
+			const balance = TokenDbSchema.safeParse(balanceRaw)
+			if (balance.success) {
+				balances.push(balance.data)
+			}
+		})
+	}
+
+	return {
+		loading: false,
+		balances,
+	}
+}
+
+async function storeBalances(waku: LightNode, address: string, balanceState: BalanceState) {
+	return await storeDocument(
+		waku,
+		'balances',
+		address,
+		balanceState.balances.map((token) => ({ ...token, amount: token.amount.toString() })),
+	)
+}
+
```

Comment:
Do you want to store balances in waku? Imho this is quite temporary and it may just complicate our life. I would either skip this for now, or implement at least reading the balances from the chain (we don't need to necessarily do the transaction sending for now).

Reply:
I agree that we should not store the balances in waku. The only reason I did this so that the balances actually change when there is a transaction. We can skip this for now.